oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -225,7 +227,14 @@ public Query baseQuery() {
                 hasTieBreaker = true;
             } else if (JCR_SCORE.equals(sortPropertyName)) {
                 fieldName = "_score";
-            } else if (indexProperties.containsKey(sortPropertyName)) {
+            } else if (indexProperties.containsKey(sortPropertyName) || elasticIndexDefinition.getDefinedRules()
+                    .stream().map(rule -> Optional.ofNullable(rule.getConfig(sortPropertyName)))
+                    .filter(Optional::isPresent)
+                    .collect(Collectors.toList()).size() > 0) {
```

Comment:
this can be optimized with (no need to transform it into a list):
```suggestion
                    .stream().map(rule -> Optional.ofNullable(rule.getConfig(sortPropertyName))).anyMatch(Optional::isPresent)) {
```